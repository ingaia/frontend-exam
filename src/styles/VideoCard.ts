import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import { LazyLoadImage } from 'react-lazy-load-image-component';

export const Item = styled(Grid)`
  &:nth-child(1) {
    margin-top: 20px;
  }

  @media (min-width: 576px) {
    &:nth-child(1) {
      margin-top: 40px;
    }
  }

  @media (min-width: 960px) {
    &:nth-child(1), &:nth-child(2) {
      margin-top: 58px;
    }
  }
`
export const Block = styled.div<{ paddingTop?: boolean }>`
  border: 1px solid var(--hillary);
  background: var(--coldGray);
  position: relative;
  cursor: pointer;


`

export const Image = styled(LazyLoadImage)`
  width: 100%;
  height: auto;
`

export const Title = styled.span`
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  width: 100%;
  background: rgba(0, 0, 0, 0.6);
  text-align: center;
  padding: 15px 0;
`
