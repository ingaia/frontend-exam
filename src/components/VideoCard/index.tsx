import React from 'react';

import * as S from '../../styles/VideoCard';

export interface Props {
  title: string,
  cover: string,
  onClick: () => void;
};

const Component = ({ title, onClick, cover }: Props) => {
  return (
    <S.Item item xs={12} md={6}>
      <S.Block onClick={onClick}>
        <S.Image src={cover} alt={title} />
        <S.Title>{title}</S.Title>
      </S.Block>
    </S.Item>
  )
};

export default Component;
