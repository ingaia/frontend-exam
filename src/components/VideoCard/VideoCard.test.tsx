import React from 'react';
import { render } from '@testing-library/react';
import VideoCard from '.';

describe('<VideoCard />', () => {
  it('should render correctly', () => {
    const { container, getByAltText } = render(
      <VideoCard title="Test component" cover="/assets/background.webp" onClick={() => {}} />
    );

    expect(getByAltText('Test component').src).toContain('/assets/background.webp');
    expect(container.firstChild).toMatchSnapshot();
  })
})
