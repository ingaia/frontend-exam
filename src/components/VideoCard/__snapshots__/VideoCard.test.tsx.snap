// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoCard /> should render correctly 1`] = `
.c0:nth-child(1) {
  margin-top: 20px;
}

.c1 {
  border: 1px solid var(--hillary);
  background: var(--coldGray);
  position: relative;
  cursor: pointer;
}

.c2 {
  width: 100%;
  height: auto;
}

.c3 {
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  width: 100%;
  background: rgba(0,0,0,0.6);
  text-align: center;
  padding: 15px 0;
}

@media (min-width:576px) {
  .c0:nth-child(1) {
    margin-top: 40px;
  }
}

@media (min-width:960px) {
  .c0:nth-child(1),
  .c0:nth-child(2) {
    margin-top: 58px;
  }
}

<div
  class="MuiGrid-root c0 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
>
  <div
    class="c1"
  >
    <img
      alt="Test component"
      class="c2"
      src="/assets/background.webp"
    />
    <span
      class="c3"
    >
      Test component
    </span>
  </div>
</div>
`;
